### DISCUSSION TITLE
wagmi contract write on sepolia

### DISCUSSION THREAD

**Discussion by CodeWithJoe2020** [wevm/wagmi#3049] [Q&A]
I have this code here

import React from 'react';
import { usePrepareContractWrite, useContractWrite } from 'wagmi';
import lottoABI from './lottoABI.json';
import { parseEther } from 'viem';

// Your component code here

function Write() {
  const { config, error } = usePrepareContractWrite({
    address: '0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3',
    abi: lottoABI, // Use the imported ABI JSON file
    functionName: 'participate',
    value:parseEther('0.01'),
    args:['0x293Bcb20584a2e0E6d1b559E25783311ec29121C']
  })
  const { data, isLoading, isSuccess, write } = useContractWrite(config)
  console.log(data)

  return (
    <>
      <button disabled={!write || isLoading} onClick={() => write?.()}>
        {isLoading ? 'Loading...' : isSuccess ? 'Success!' : 'Bought'}
      </button>
      {error && (
        <div>An error occurred preparing the transaction: {error.message}</div>
      )}
    </>
  );
}
export default Write;

this function is suppsed to call a function on a smart contract with ETH value and an referral address

but when i run this in my nextjs project i see this error
ContractFunctionExecutionError: HTTP request failed.

URL: https://rpc.sepolia.org
Request body: {"method":"eth_call","params":[{"from":"0x293Bcb20584a2e0E6d1b559E25783311ec29121C","data":"0xb91038c7000000000000000000000000293bcb20584a2e0e6d1b559e25783311ec29121c","to":"0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3"},"latest"]}
 
Raw Call Arguments:
  from:  0x293Bcb20584a2e0E6d1b559E25783311ec29121C
  to:    0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3
  data:  0xb91038c7000000000000000000000000293bcb20584a2e0e6d1b559e25783311ec29121c
 
Contract Call:
  address:   0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3
  function:  participate(address _referrer)
  args:                 (0x293Bcb20584a2e0E6d1b559E25783311ec29121C)
  sender:    0x293Bcb20584a2e0E6d1b559E25783311ec29121C

Docs: https://viem.sh/docs/contract/simulateContract.html
Details: NetworkError when attempting to fetch resource.

and i also get this error
An error occurred preparing the transaction: Execution reverted for an unknown reason. Raw Call Arguments: from: 0x293Bcb20584a2e0E6d1b559E25783311ec29121C to: 0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3 value: 0.01 SEP data: 0xb91038c7000000000000000000000000293bcb20584a2e0e6d1b559e25783311ec29121c Contract Call: address: 0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3 function: participate(address _referrer) args: (0x293Bcb20584a2e0E6d1b559E25783311ec29121C) sender: 0x293Bcb20584a2e0E6d1b559E25783311ec29121C Docs: https://viem.sh/docs/contract/simulateContract.html Details: execution reverted Version: viem@0.3.39

im on sepolia testnet
does anyone know what i need to do to fix this?

**Comment by CodeWithJoe2020** [comment]
can anyone please answer my question?

**Reply by GAEAlimited** [reply]
The error message you're encountering suggests that there is an issue when trying to execute a contract function on the Sepolia testnet using the Viem library. There could be several reasons for this error. Let's try to diagnose and fix the issue step by step:

HTTP Request Failure: The first error message indicates that an HTTP request to the Sepolia testnet failed. This could be due to a network issue or Sepolia's server being down. You might want to ensure that the Sepolia testnet is running and accessible. Try accessing the Sepolia testnet URL directly in your browser to see if it's online.

Execution Reverted: The second error message indicates that the contract function execution was reverted for an unknown reason. This typically happens when there is an issue with the contract code itself or the parameters you are passing.

Check the contract code to ensure that the participate function is implemented correctly.

Verify that the contract at address 0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3 is deployed and is the correct contract.

Ensure that the lottoABI you imported matches the ABI of the deployed contract.

Gas and ETH Balance: Make sure that the account with the address 0x293Bcb20584a2e0E6d1b559E25783311ec29121C has a sufficient ETH balance to cover the gas cost of the transaction.

Check Viem and Wagmi Versions: Ensure that you are using compatible versions of Viem and Wagmi. Sometimes, updates to these libraries can introduce breaking changes or compatibility issues.

Network Configuration: Check if you need to specify a custom network configuration for the Sepolia testnet in your code. Make sure that the network configuration matches the Sepolia testnet's endpoints and settings.

Debugging: You can try to add additional logging and error handling to your code to get more detailed information about the issue. For example, you can log the transaction hash and event logs to see if there are any additional error messages from the contract.

Contact Sepolia Support: If none of the above steps resolve the issue, consider reaching out to Sepolia's support or community forums for assistance. They might have specific guidance or insights into the Sepolia testnet's behavior.

Remember to double-check your contract's code, ABI, and network settings to ensure everything is set up correctly. Contract execution errors can be challenging to debug, so a careful review of each component is essential.
